README.md

```diff
@@ -162,6 +162,53 @@ spec:
       method: request.method | "get"
 ```
 
+#### OpenID Connect Pattern
+To use the *OpenID Connect authentication pattern*, you should use the `properties` value on the `subject` field to set `client_id`, and **optionally** `app_key`.
+
+Manipulation of this object can be done in using the methods described previously.
+An example configuration is shown below. Here the client identifier (application ID) is parsed from the JWT under the label "azp". Modify this as desired.
+
+```yaml
+apiVersion: "config.istio.io/v1alpha2"
+kind: instance
+metadata:
+  name: threescale-authorization
+  namespace: istio-system
+spec:
+  template: threescale-authorization
+  params:
+    subject:
+        app_key: request.query_params["app_key"] | request.headers["x-app-key"] | ""
+        client_id: request.auth.claims["azp"] | ""
+    action:
+      path: request.url_path
+      method: request.method | "get"
+```
+
+For this integration to work correctly. OpenID configuration must still be done in 3scale in order for the client to be created in the IdP.
+For the service the user wants to protect, they should create [end-user authentication](https://istio.io/help/ops/security/end-user-auth/) in the same namespace as that service.
```

Comment:
we might want to generate this as well in the CLI tool?

Reply:
Perhaps. Could also argue that this is entirely different object and needs to be created by the user in their namesake. I think the benefit of us generation the others comes from the fact that we have defined the spec for those CR’s. In the case however, this object is well defined by Istio. I’m not opposed to generating it in the end of the day, but perhaps for discussion and as before, a different PR. WDYT?

Reply:
a different PR and a different flag to generate just that (assuming no one wants to generate the whole set of templates + this in a single invocation to then pipe it to `oc`) - in fact, you are right that this is very different, and indeed could mean we would need to rework the CLI to accept subcommands (ie. generate service config vs generate authn/OIDC config)

Reply:
Yes was thinking something along the same if we go to do it.